server/src/processing/LinkDetectionModule/LinkDetectionModule.ts

```diff
@@ -121,6 +126,11 @@ export class LinkDetectionModule extends Module {
   */
   private async extractLinksFromMetadata(file: string): Promise<JSON[]> {
     const annots = [];
+    const fileType: { ext: string; mime: string } = filetype(fs.readFileSync(file));
```

Comment:
I think this validation has to be done in first lines of main module.

I will move these lines to line 36.
`public async main(doc: Document): Promise<Document> {
          const fileType: { ext: string; mime: string } = filetype(fs.readFileSync(doc.inputFile));
          if (fileType === null || fileType.ext !== 'pdf') {
                    logger.warn(`Warning: The input file ${file} is not a PDF bla bla bla not fetching meta information for link detection..`);
                    return doc;
    }
`